store/names.js

```diff
@@ -1,23 +1,33 @@
+import { loadNextPage } from './utils'
+
 export const state = () => ({
-  names: []
+  names: [],
+  nextPageUrl: ''
 })
 
 export const mutations = {
   setNames (state, names) {
-    state.names = [...state.names, ...names]
+    state.names = [...state.names, ...names.data]
+    state.nextPageUrl = names.next
   }
 }
 
 export const actions = {
-  getNames: async function ({ rootGetters: { middleware }, commit }, { page, limit }) {
+  getLatestNames: async function ({ rootGetters: { middleware }, state: { nextPageUrl }, commit }, { page, limit }) {
     try {
+      if (nextPageUrl) return
       const names = await middleware.getAllNames({ page, limit })
-      commit('setNames', names.data)
+      commit('setNames', names)
     } catch (e) {
       console.log(e)
       commit('catchError', 'Error', { root: true })
     }
   },
+  getMoreNames: async function ({ state: { nextPageUrl }, commit }) {
+    if (!nextPageUrl) return
+    const names = await loadNextPage(nextPageUrl)
+    commit('setNames', names)
+  },
```

Comment:
Would be interesting to use [generators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator) here, but I think they can't be passed over network :(